### PR TITLE
image-viewer

### PR DESCRIPTION
**Pull request by bwalsh**
This PR:
* adds helm chart and revproxy config for image-viewer, viv
* see [wiki](https://wiki.ohsu.edu/display/KEDTW/Frontend+Framework+Setup?focusedCommentId=336036037#comment-336036037) for setup for FEF local dev

### CODE REVIEW

**Comment by matthewpeterkort** [comment]
Looks good so far. 

Add a service.yaml for local-image-viewer  similar to how https://github.com/ACED-IDP/gen3-helm/blob/ohsu-develop/helm/fence/templates/service.yaml#L1-L15 and others are implemented so that don't have to manually run the expose command

File: helm/image-viewer/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# To get image-viewer to be discoverable to other microservices run the following command
+
+kc expose deployment local-image-viewer
```
Comment (matthewpeterkort) [inline helm/image-viewer/README.md:3]: add a service file to avoid this

**Review by matthewpeterkort — CHANGES_REQUESTED** [review]
see https://github.com/ACED-IDP/gen3-helm/pull/69#issuecomment-2400569362

File: helm/image-viewer/README.md (hunk 1 above)
Comment (bwalsh) [reply helm/image-viewer/README.md:3]: @matthewpeterkort  thanks for the review.  I actually have a service.yaml but this line is stopping it from being added?

https://github.com/ACED-IDP/gen3-helm/blame/ohsu-develop/.gitignore#L5

Comment (bwalsh) [reply helm/image-viewer/README.md:3]: I just used `git add -f` to add it

Comment (matthewpeterkort) [reply helm/image-viewer/README.md:3]: Ok thank you. Yeah that seems odd. Might be good to remove that.

File: helm/image-viewer/values.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# Default values for image-viewer.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
+serviceName: image-viewer-service
+
+image:
+  repository: gen3-image-viewer
```
Comment (matthewpeterkort) [inline helm/image-viewer/values.yaml]: This is causing repo fetch errors I assume this is overwritten by the image-viewer repo specified in the secrets file?

**Comment by matthewpeterkort** [comment]
exec into running pod, build venv, run pytests. Pytests pass inside pod except `test_base_url_from_env_file` errors with 

```
# Assert that the base_url comes from the .env file
        expected_base_url = 'https://env-file-url.com/objects/'
        assert response.status_code == 307
>       assert response.headers["location"].startswith(expected_base_url)
E       AssertionError: assert False
```
and [2024-10-08 23:32:33,236][WARNING] Unable to find refresh_file

File: helm/image-viewer/values.yaml (hunk 2 above)
Comment (bwalsh) [reply helm/image-viewer/values.yaml]: Exactly.  The stanza I have in my Secrets/values.yaml:

```
image-viewer:
  serviceName: image-viewer-service
  image:
    repository: quay.io/ohsu-comp-bio/gen3-image-viewer
    pullPolicy: Always
    tag: "latest"
```

**Comment by bwalsh** [comment]
@matthewpeterkort All tests now pass in pod
```
root@image-viewer-deployment-57fb9cbcbc-qhvvm:/app/image_viewer# git log --oneline | head -1
3bdb3d7 include .env file for test
root@image-viewer-deployment-57fb9cbcbc-qhvvm:/app/image_viewer# pytest tests/
........                                                                                                                                                   [100%]
8 passed in 6.38s
```

**Comment by matthewpeterkort** [comment]
Working locally. Going to test this on cbds-dev

**Comment by matthewpeterkort** [comment]
tested and running on cbds-dev LGTM

**Review by matthewpeterkort — DISMISSED** [review]

**Review by matthewpeterkort — DISMISSED** [review]

**Review by lbeckman314 — APPROVED** [review]

**Pull request merged**